### PR TITLE
support xgboost regression model

### LINKED ISSUE #795: Using XGBoost to make regression prediction but using 'argmax' in classifier instead.
**Description**
Using XGBoost to make regression prediction but using 'argmax' in classifier instead.

**Reproduction Steps**
```
%%sqlflow
SELECT *
FROM datamap_test.ds_xgboost_test
PREDICT ds_xgboost_predict.duration
USING datamap_test.ds_xgboost_model ;
```

**Expected Behavior**
Make right regression.

**Screenshots**
```
[15:47:58] 8913x57 matrix with 508041 entries loaded from predict.txt

Traceback (most recent call last):

  File "<stdin>", line 618, in <module>

  File "/home/luban/miniconda/envs/sqlflow-dev-py36/lib/python3.6/site-packages/numpy/core/fromnumeric.py", line 1103, in argmax

    return _wrapfunc(a, 'argmax', axis=axis, out=out)

  File "/home/luban/miniconda/envs/sqlflow-dev-py36/lib/python3.6/site-packages/numpy/core/fromnumeric.py", line 56, in _wrapfunc

    return getattr(obj, method)(*args, **kwds)

numpy.AxisError: axis 1 is out of bounds for array of dimension 1

StatusCode.UNKNOWN
exit status 1
```

### PR DESCRIPTION
**Pull request by Yancey0623**
fixed #795

### CODE REVIEW

File: cmd/sqlflowserver/main_test.go [hunk 1]
```diff
@@ -1000,3 +1001,68 @@ FROM housing.predict LIMIT 5;`)
...
+func CaseTrainXGBoostRegression(t *testing.T) {
+	a := assert.New(t)
+	trainSQL := fmt.Sprintf(`
+SELECT *
+FROM housing.train
+TRAIN xgboost.gbtree
+WITH
+		objective="reg:squarederror",
+		train.num_boost_round = 30
+		COLUMN f1,f2,f3,f4,f5,f6,f7,f8,f9,f10,f11,f12,f13
```
Comment (weiguoz) [inline cmd/sqlflowserver/main_test.go:1015]: indent

File: sql/template_xgboost.go [hunk 2]
```diff
@@ -134,8 +134,11 @@ dpred = xgb_dataset('predict.txt', """{{.PredictionDatasetSQL}}""")
 bst = xgb.Booster({'nthread': 4})  # init model
 bst.load_model("{{.Save}}")  # load data
 preds = bst.predict(dpred)
-# TODO(typhoonzero): regression models may have different behavior
-pred_classes = np.argmax(np.array(preds), axis=1)
+
+# TODO(Yancey1989): using the train parameters to decide regressoin model or classifier model
```
Comment (tonyyang-svail) [inline sql/template_xgboost.go:138]: how to get the train parameters? Should we add `TrainIR` to `PredictIR`?

**Review by tonyyang-svail — APPROVED** [review]
LGTM.

File: sql/template_xgboost.go (hunk 2 above)
Comment (Yancey0623) [reply sql/template_xgboost.go:138]: Maybe we can save the TrainClause in the model struct: 
https://github.com/sql-machine-learning/sqlflow/blob/5ae82c045024e632a9d5eb898225c631187d688f/sql/model.go#L27-L30

And the prediction job can resolve the TrainClause to the parameters. And I think `PredictIR` is a good idea, do you want to resolve the Predict SQL and saved model into `PredictIR`?

File: cmd/sqlflowserver/main_test.go (hunk 1 above)
Comment (Yancey0623) [reply cmd/sqlflowserver/main_test.go:1015]: Thanks, will fix it in the next PR.

File: sql/template_xgboost.go [hunk 3]
```diff
@@ -134,8 +134,11 @@ dpred = xgb_dataset('predict.txt', """{{.PredictionDatasetSQL}}""")
...
+if len(preds.shape) == 2:
```
Comment (typhoonzero) [inline sql/template_xgboost.go:139]: Should find a more accurate way to determine whether the model is a regression model.

**Pull request merged**